MobileBuy/buy/src/androidTest/java/com/shopify/buy/service/BuyTest.java

```diff
@@ -226,13 +226,30 @@ public void testFetchingShippingRatesWithInvalidCheckout() throws InterruptedExc
     public void testCheckoutFlowUsingCreditCard() throws InterruptedException {
         createValidCheckout();
         fetchShippingRates(HttpStatus.SC_OK);
-        updateCheckout();
+        setShippingRate();
         addCreditCardToCheckout();
         completeCheckout();
         pollCheckoutCompletionStatus();
         getCheckout();
     }
 
+    public void testChangedShippingAddress() throws InterruptedException {
+        createValidCheckout();
+        fetchShippingRates(HttpStatus.SC_OK);
+        setShippingRate();
+
+        Address address = checkout.getShippingAddress();
+        address.setCity("Toronto");
+
```

Comment:
Not sure if the below functions are needed. At this point, just check if the shipping rate has been invalidated?